src/cryptonote_core/tx_pool.cpp

```diff
@@ -913,6 +913,34 @@ namespace cryptonote
     }, false, category);
   }
   //------------------------------------------------------------------
+  void tx_memory_pool::get_block_template_backlog(std::vector<tx_block_template_backlog_entry>& backlog, bool include_sensitive) const
+  {
+    CRITICAL_REGION_LOCAL(m_transactions_lock);
+    CRITICAL_REGION_LOCAL1(m_blockchain);
+    const relay_category category = include_sensitive ? relay_category::all : relay_category::broadcasted;
+    backlog.reserve(m_blockchain.get_txpool_tx_count(include_sensitive));
+    cryptonote::blobdata tmp_blob;
+    txpool_tx_meta_t tmp_meta;
+    m_blockchain.for_all_txpool_txes([this, &backlog, &tmp_blob, &tmp_meta](const crypto::hash &txid, const txpool_tx_meta_t &meta, const cryptonote::blobdata_ref *bd){
+      tmp_blob = m_blockchain.get_txpool_tx_blob(txid, relay_category::all);
```

Comment:
If the `false` on line 941 is changed to `true` then this call is unnecessary because `bd` is a pointer to the transaction blob. This should be a lot more efficient.